### PR TITLE
Task av/j168 add comments timelines near the audio

### PR DESCRIPTION
**Pull request by artem-totality**

### CODE REVIEW

File: frontend/src/components/episode/components/timeline-item/timeline-item.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  comment,
+  dimensions,
+  duration,
+  handleGoToTimestamp,
+}) => {
+  const commentOffset =
+    ((dimensions.offset +
+      (comment.timestamp / duration) * dimensions.progressBarWidth) /
+      dimensions.playerContainerWidth) *
+    100;
```
Comment (what1s1ove) [inline frontend/src/components/episode/components/timeline-item/timeline-item.tsx]: move this calculation to the function `frontend/src/components/episode/components/timeline-item/helpers`

File: frontend/src/components/common/player/player.tsx [hunk 2]
```diff
@@ -28,7 +29,9 @@ import { ReactComponent as MuteIcon } from 'assets/img/player/mute.svg';
 import { ReactComponent as RateScaleIcon } from 'assets/img/player/rateScale.svg';
 import { ReactComponent as RatePointerIcon } from 'assets/img/player/ratePointer.svg';
 import 'react-h5-audio-player/lib/styles.css';
+import { player as playerActions } from 'store/actions';
 import styles from './styles.module.scss';
+import { DataStatus } from 'common/enums/enums';
```
Comment (what1s1ove) [inline frontend/src/components/common/player/player.tsx]: keep import order

File: frontend/src/components/common/player/player.tsx [hunk 3]
```diff
@@ -95,6 +101,15 @@ const Player = forwardRef<Ref, Props>(
         }}
         onClickNext={onClickNext}
         onClickPrevious={onClickPrevious}
+        onLoadStart={(): void => {
+          dispatch(playerActions.setPlayerStatus(DataStatus.PENDING));
```
Comment (what1s1ove) [inline frontend/src/components/common/player/player.tsx]: remove anonymous functions from the `jsx`

File: frontend/src/components/episode/components/comments-timeline/comments-timeline.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    duration,
+  };
+
+  return (
+    <div className={styles.timeline}>
+      {comments.map((item) => (
+        <TimelineItem
+          key={item.id}
+          comment={item}
+          {...props}
```
Comment (what1s1ove) [inline frontend/src/components/episode/components/comments-timeline/comments-timeline.tsx]: avoid it

File: frontend/src/components/episode/components/comments-timeline/comments-timeline.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+          handleGoToTimestamp={(): void => {
```
Comment (what1s1ove) [inline frontend/src/components/episode/components/comments-timeline/comments-timeline.tsx]: check the **A9** criteria
Remove anonymous functions from the jsx

File: frontend/src/components/episode/components/timeline-item/timeline-item.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+  return (
+    <div
+      className={styles.timelineItem}
+      style={{ left: `${commentOffset}%` }}
+      onClick={handleGoToTimestamp}
```
Comment (what1s1ove) [inline frontend/src/components/episode/components/timeline-item/timeline-item.tsx]: don't use `onClick` on `div`. Use `button` instead

File: frontend/src/components/episode/episode.tsx [hunk 7]
```diff
@@ -38,6 +39,7 @@ const Episode: React.FC = () => {
       comments: episode.comments,
       user: auth.user,
       podcast: episode.podcast,
+
```
Comment (what1s1ove) [inline frontend/src/components/episode/episode.tsx]:
```suggestion
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by JuliaDzyuba — DISMISSED** [review]

File: frontend/src/components/common/player/player.tsx [hunk 8]
```diff
@@ -32,6 +33,7 @@ import styles from './styles.module.scss';
 
 type Props = {
   src: string;
+  setPlayerStatus: React.Dispatch<React.SetStateAction<DataStatus>>;
```
Comment (what1s1ove) [inline frontend/src/components/common/player/player.tsx]: check **A9** criteria

File: frontend/src/components/episode/components/comments-timeline/comments-timeline.tsx [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+type Props = {
+  comments: TComment[];
+  playerRef: React.MutableRefObject<H5AudioPlayer | null>;
+  playerContainerRef: React.RefObject<HTMLDivElement>;
+};
+
+const CommentsTimeline: React.FC<Props> = ({
+  comments,
+  playerRef,
+  playerContainerRef,
```
Comment (what1s1ove) [inline frontend/src/components/episode/components/comments-timeline/comments-timeline.tsx]:
```suggestion
```
This component shouldn't know anything about player ref

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by yevhenprybytok — DISMISSED** [review]

File: frontend/src/components/common/player/player.tsx [hunk 10]
```diff
@@ -32,6 +34,7 @@ import styles from './styles.module.scss';
 
 type Props = {
   src: string;
+  handleSetPlayerStatus: React.Dispatch<React.SetStateAction<DataStatus>>;
```
Comment (what1s1ove) [inline frontend/src/components/common/player/player.tsx]: Check **A8** criteria

File: frontend/src/components/common/player/player.tsx [hunk 11]
```diff
@@ -68,6 +73,14 @@ const Player = forwardRef<Ref, Props>(
 
         return DEFAULT_CURRENT_TIME;
       },
+      getPodcastDuration: (): number => {
+        if (playerRef.current && playerRef.current.audio.current) {
```
Comment (what1s1ove) [inline frontend/src/components/common/player/player.tsx]:
```suggestion
        if (playerRef.current?.audio.current) {
```

File: frontend/src/components/episode/components/timeline-item/timeline-item.tsx [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+    duration,
+    comment.timestamp,
+    dimensions,
+  );
+
+  return (
+    <button
+      className={styles.timelineItem}
+      style={{ left: `${commentOffset}%` }}
+      onClick={handleGoToItemTimestamp(comment.timestamp)}
```
Comment (what1s1ove) [inline frontend/src/components/episode/components/timeline-item/timeline-item.tsx]: remove anonymous functions from jsx

File: frontend/src/components/episode/components/timeline-item/timeline-item.tsx [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+  handleGoToItemTimestamp,
+}) => {
+  const commentOffset = getCommentOffset(
+    duration,
+    comment.timestamp,
+    dimensions,
+  );
+
+  return (
+    <button
```
Comment (what1s1ove) [inline frontend/src/components/episode/components/timeline-item/timeline-item.tsx]:
```suggestion
    <li>
     <button className={styles.timelineItem}>
     //
     </li>
```

File: frontend/src/components/episode/helpers/get-comments-timeline-dimensions/get-comments-timeline-dimensions.helper.ts [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+
+  const offset = progressBarLeft - playerContainerLeft;
+
+  const dimensions = {
+    offset,
+    progressBarWidth,
+    playerContainerWidth,
+  };
+
+  return dimensions;
```
Comment (what1s1ove) [inline frontend/src/components/episode/helpers/get-comments-timeline-dimensions/get-comments-timeline-dimensions.helper.ts]:
```suggestion
  return {
    offset,
    progressBarWidth,
    playerContainerWidth,
  };
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/player/player.tsx (hunk 10 above)
Comment (what1s1ove) [reply frontend/src/components/common/player/player.tsx]: don't fixed

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**